src/main/java/org/codehaus/mojo/license/AggregatorAddThirdPartyMojo.java

```diff
@@ -203,11 +204,30 @@ protected void doAction()
         }
         if ( groupId == null )
         {
-            throw new IllegalStateException( "Can't find license-maven-plugin" );
+            try
+            {
+                final PluginDescriptor pd = ( PluginDescriptor ) getPluginContext().get( "pluginDescriptor" );
+                groupId = pd.getGroupId();
+                artifactId = pd.getArtifactId();
+                version = pd.getVersion();
```

Comment:
Isn't this actually the preferred way to get the `g:a:v` of the current plugin? If so, I guess we can remove the attempts above (startig at https://github.com/mojohaus/license-maven-plugin/pull/348/files#diff-38b03aa7a6c189ea0f7e814313a14682R182 ) to get it via ` project.getPluginArtifactMap()` and `project.getPluginManagement().getPluginsAsMap().get( "org.codehaus.mojo:license-maven-plugin" )` ?

Reply:
On premise, never touch a running system. I just didn't want to throw that away. I also just updated an existing IT by removing the build dep. I don't think the fix needs it's own IT.